libvast/vast/format/bgpdumpbinary.hpp

```diff
@@ -0,0 +1,1455 @@
+#ifndef VAST_FORMAT_BGPDUMPBINARY_HPP
+#define VAST_FORMAT_BGPDUMPBINARY_HPP
+
+#include <istream>
+#include <iostream>
+#include "vast/logger.hpp"
+#include "vast/schema.hpp"
+#include "vast/time.hpp"
+#include "vast/date.hpp"
+#include "vast/address.hpp"
+#include "vast/subnet.hpp"
+#include "vast/data.hpp"
+#include "vast/concept/parseable/core.hpp"
+#include "vast/concept/parseable/string.hpp"
+#include "vast/concept/parseable/numeric/byte.hpp"
+#include "vast/concept/parseable/vast/address.hpp"
+#include "vast/concept/parseable/vast/time.hpp"
+#include "vast/concept/parseable/vast/subnet.hpp"
+#include "vast/concept/parseable/vast/data.hpp"
+#include "vast/concept/printable/vast/address.hpp"
+#include "vast/concept/printable/vast/subnet.hpp"
+#include "vast/concept/printable/vast/data.hpp"
+#include "vast/concept/printable/string.hpp"
+#include "vast/concept/printable/to_string.hpp"
+#include "vast/concept/printable/std/chrono.hpp"
+
+#include "vast/format/reader.hpp"
+
+namespace vast {
+namespace format {
+namespace bgpdumpbinary {
+
+struct form {
+  timestamp ts;
+  count bgp_type;
+  count type;
+  count subtype;
+  count interface_index;
+  count addr_family;
+  count old_state;
+  count new_state;
+  count bgp_length;
+  count length;
+  count pasnr;
+  count med;
+  count local_pref;
+  count version;
+  count my_autonomous_system;
+  count hold_time;
+  count bgp_id;
+  count error_code;
+  count error_subcode;
+  std::string msg_type;
+  std::string origin;
+  std::string as_path_orded;
+  std::string community;
+  std::string atomic_aggregate;
+  address peer_ip_v4;
+  address peer_ip_v6;
+  address nexthop_v4;
+  address nexthop_v6;
+  std::vector<data> as_path;
+  std::vector<subnet> prefix_v4;
+  std::vector<subnet> prefix_v6;
+  std::tuple<count, address> aggregator;
+
+  /* extended fields */
+  uint16_t wd_rts_len;
+  bool paket_error;
+  std::vector<form> rib_entries;
+};
+
+/// A parser that reading bgp messages from the MRT files.
+struct bgpdumpbinary_parser : parser<bgpdumpbinary_parser> {
+  using attribute = event;
+
+  bgpdumpbinary_parser();
+
+  template <class Iterator>
+  bool parse(Iterator &f, Iterator const &l, unused_type) const {
+    if(f >= l)
+      return false;
+    else
+      return true;

```

Comment:
While this really should be `return f >= l`, it's not the correct implementation. The idea of the overload with `unused_type` is that it should only parse the input (e.g., to conform that it is well-defined) without populating the attribute of the parser.